libs/core/type_support/tests/unit/is_trivially_relocatable.cpp

```diff
@@ -0,0 +1,155 @@
+};
+
+struct non_assignable_but_trivially_copyable
+{
+    int i;
+    explicit non_assignable_but_trivially_copyable(int);
+    non_assignable_but_trivially_copyable(
+        non_assignable_but_trivially_copyable&&) = default;
+    non_assignable_but_trivially_copyable operator=(
+        non_assignable_but_trivially_copyable&&) = delete;
+    ~non_assignable_but_trivially_copyable() = default;
+};
+
+static_assert(hpx::is_trivially_relocatable_v<empty>);
+static_assert(hpx::is_trivially_relocatable_v<non_empty_but_trivial>);
+static_assert(
+    hpx::is_trivially_relocatable_v<non_trivial_but_trivially_copyable>);
+static_assert(
+    hpx::is_trivially_relocatable_v<move_only_but_trivially_copyable>);
+static_assert(
+    hpx::is_trivially_relocatable_v<non_assignable_but_trivially_copyable>);
+
+// Has non trivial copy constructor
+struct not_trivially_copyable_1
+{
+    not_trivially_copyable_1();
+    not_trivially_copyable_1(not_trivially_copyable_1 const&);
+    not_trivially_copyable_1& operator=(
+        not_trivially_copyable_1 const&) = default;
+    ~not_trivially_copyable_1() = default;
+};
+
+// Has non trivial copy assignment
+struct not_trivially_copyable_2
+{
+    not_trivially_copyable_2();
+    not_trivially_copyable_2(not_trivially_copyable_2 const&) = default;
+    not_trivially_copyable_2& operator=(not_trivially_copyable_2 const&);
+    ~not_trivially_copyable_2() = default;
+};
+
+// Has non trivial destructor
+struct not_trivially_copyable_3
+{
+    not_trivially_copyable_3();
+    not_trivially_copyable_3(not_trivially_copyable_3 const&) = default;
+    not_trivially_copyable_3& operator=(
+        not_trivially_copyable_3 const&) = default;
+    ~not_trivially_copyable_3();
+};
+
+// Non trivially copyable types are not trivially relocatable
+// Unless they are explicitly declared as such
+static_assert(!hpx::is_trivially_relocatable_v<not_trivially_copyable_1>);
+static_assert(!hpx::is_trivially_relocatable_v<not_trivially_copyable_2>);
+static_assert(!hpx::is_trivially_relocatable_v<not_trivially_copyable_2>);
+
+// Not trivially copyable
+struct explicitly_trivially_relocatable_1
+{
+    explicitly_trivially_relocatable_1() = default;
+    explicitly_trivially_relocatable_1(
+        explicitly_trivially_relocatable_1 const&);
+    explicitly_trivially_relocatable_1& operator=(
+        explicitly_trivially_relocatable_1 const&);
+    ~explicitly_trivially_relocatable_1() = default;
+};
+
+// Has non trivial copy assignment
+struct explicitly_trivially_relocatable_2
+{
+    explicitly_trivially_relocatable_2() = default;
+    explicitly_trivially_relocatable_2(explicitly_trivially_relocatable_2&&);
+    explicitly_trivially_relocatable_2& operator=(
+        explicitly_trivially_relocatable_2&&);
+    ~explicitly_trivially_relocatable_2() = default;
+};
+
+HPX_DECLARE_TRIVIALLY_RELOCATABLE(explicitly_trivially_relocatable_1);
+HPX_DECLARE_TRIVIALLY_RELOCATABLE(explicitly_trivially_relocatable_2);
+
```

Comment:
Please add a test for a trivially copyable type that is also explicitly marked as trivially relocatable.